test/integration/arrakisUniswapV3AmmAdapter.spec.ts

```diff
@@ -0,0 +1,506 @@
+import "module-alias/register";
+
+import { network } from "hardhat";
+import { BigNumber } from "ethers";
+import { ether } from "@utils/index";
+import { Account } from "@utils/test/types";
+import { Address } from "@utils/types";
+import {
+  MAX_UINT_256,
+  ADDRESS_ZERO,
+  ZERO,
+} from "@utils/constants";
+import { SetToken, AmmModule, ArrakisUniswapV3AmmAdapter } from "@utils/contracts";
+import DeployHelper from "@utils/deploys";
+import {
+  addSnapshotBeforeRestoreAfterEach,
+  getAccounts,
+  getSystemFixture,
+  getUniswapV3Fixture,
+  getArrakisV1Fixture,
+  getWaffleExpect
+} from "@utils/test/index";
+import { SystemFixture, UniswapV3Fixture, ArrakisV1Fixture } from "@utils/fixtures";
+
+const expect = getWaffleExpect();
+
+describe("ArrakisUniswapV3AmmAdapter Integration [ @forked-mainnet ]", () => {
+  let owner: Account;
+  let deployer: DeployHelper;
+  let setup: SystemFixture;
+  let uniswapV3Setup: UniswapV3Fixture;
+  let arrakisV1Setup: ArrakisV1Fixture;
+  let ammModule: AmmModule;
+  let arrakisUniswapV3AmmAdapter: ArrakisUniswapV3AmmAdapter;
+  let arrakisUniswapV3AmmAdapterName: string;
+
+  before(async () => {
+    await network.provider.request({
+      method: "hardhat_reset",
+      params: [
+        {
+          forking: {
```

Comment:
I think this hardhat reset is causing the (seemingly unrelated) test failures in the other tests.

We have two options here:
1. Resetting to the state after running these tests to the previously pinned block number.
2. Adjusting the other tests to work on this block number as well and then updating it for all tests in the hardhat config.  I can support here if needed. 

I'm a bit conflicted on which is better, but I think long term it might be nice to have different tests running against different block numbers so that we don't have to continously update some of the tests which assume specific market conditions / product states.

Reply:
Yet another test is to set to update the blocknumber in hardhat-config and then do this "hardhat_reset" to the old blocknumber for all the test suites that are failing on the new block.

Reply:
> Yet another option to update the blocknumber in hardhat-config and then do this "hardhat_reset" to the old blocknumber for all the test suites that are failing on the new block. I think this is probably the best solution as it should be the individual test suite's "responsibility" to ensure it is running agains the right block if it is sensitive to changes in market conditions / product composition etc.

hmm I thought that `addSnapshotBeforeRestoreAfterEach` would handle setting up a "resetting" to the snapshot after, but maybe its doing the snapshot after the fork which messes things up moving forward. I'll double check my understanding around this flow and likely try to implement this solution.

Reply:
This is a great hint towards a solution. I think I can rework the before hook such that from a general level, the network is reset to the fork block for every test as expected.